src/services/commit/useCommitYaml.tsx

```diff
@@ -0,0 +1,123 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+} from 'services/repo'
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+import A from 'ui/A'
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  commit: z
+    .object({
+      commitid: z.string(),
+      yaml: z.string().nullable(),
+    })
+    .nullable(),
+})
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repository: z
+        .discriminatedUnion('__typename', [
+          RepositorySchema,
+          RepoNotFoundErrorSchema,
+          RepoOwnerNotActivatedErrorSchema,
+        ])
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const query = `
+query CommitYaml($owner: String!, $repo: String!, $commitid: String!) {
+  owner(username: $owner) {
+    repository: repository(name: $repo) {
+      __typename
+      ... on Repository {
+        commit(id: $commitid) {
+          commitid
+          yaml
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+interface UseCommitYamlArgs {
+  provider: string
+  owner: string
+  repo: string
+  commitid: string
+}
+
+export function useCommitYaml({
+  provider,
+  owner,
+  repo,
+  commitid,
+}: UseCommitYamlArgs) {
+  return useQuery({
+    queryKey: ['commit-yaml', provider, owner, repo, commitid],
```

Comment:
nit: I think this is the first queryKey I've seen with a "-" in it, maybe we can remove?